public/app/features/dashboard/state/actions.ts

```diff
@@ -119,10 +121,22 @@ export function removeDashboard(uri: string): ThunkResult<void> {
 
 export function initDashboardPanel(panel: PanelModel): ThunkResult<void> {
   return async (dispatch, getStore) => {
-    let plugin = getStore().plugins.panels[panel.type];
+    let pluginToLoad = panel.type;
+
+    const isDeprecated = isDeprecatedPanel(panel.type);
+
+    if (isDeprecated) {
+      pluginToLoad = DEPRECATED_PANELS[panel.type](panel);
+    }
+
+    let plugin = getStore().plugins.panels[pluginToLoad];
 
     if (!plugin) {
-      plugin = await dispatch(loadPanelPlugin(panel.type));
+      plugin = await dispatch(loadPanelPlugin(pluginToLoad));
+    }
+
+    if (isDeprecated) {
+      dispatch(changePanelPlugin(panel, pluginToLoad));
```

Comment:
Not sure I understand this condition, you change plugin but you already switched pluginToLoad to the new panel

Reply:
oh wait now I undersand

Reply:
think you need to add an await here as changePanelPlugin is async